### PR TITLE
Get immediate children of an AST Node

### PR DESCRIPTION
**Pull request by TilakMaddy**
If I have to capture `outerX` and `outerY` but exclude `inner` stuff, you can do it with this PR

```

contract AnotherOne {

    uint outerX;
    uint outerY;

    function setOwner(address _owner) external {
        uint256 inner = 1;
        uint256 innerinner = 42;
        
    }

    function someOtherImportantThing() external {
        // do something important
        uint256 innerinner = 42;
        if (innerinner == 34) {
            uint256 ret = 1;
        }
    }

}
```

As usual, see the comments - it points out the example and the logic function

### CODE REVIEW

File: aderyn_core/src/context/browser/children.rs [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        )
+    }
+}
+impl GetImmediateChildren for WhileStatement {
+    fn immediate_children(&self, context: &WorkspaceContext) -> Option<Vec<ASTNode>> {
+        get_children_of_node(&ASTNode::WhileStatement(self.clone()), context)
+    }
+}
+
+pub fn get_children_of_node(node: &ASTNode, context: &WorkspaceContext) -> Option<Vec<ASTNode>> {
```
Comment (TilakMaddy) [inline aderyn_core/src/context/browser/children.rs:233]: Function at the heart of the PR

File: aderyn_core/src/detect/experimental/demo_children.rs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+impl IssueDetector for ChildrenDemoDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        /* The following code would capture lines
+           {40, 41} In ParentChainContract.sol
+           (This is just the outer variables}
+        */
+        for cd in context.contract_definitions.keys() {
+            if cd.name.contains("AnotherOne") {
+                for child in cd.immediate_children(context).unwrap() {
```
Comment (TilakMaddy) [inline aderyn_core/src/detect/experimental/demo_children.rs:38]: Boom! This is how we can loop through immediate children of an AST node : )

**Comment by TilakMaddy** [comment]
Now we can do BFS traveversal on the ast !

File: aderyn_core/src/detect/experimental/demo_children.rs (hunk 2 above)
Comment (TilakMaddy) [reply aderyn_core/src/detect/experimental/demo_children.rs:38]: The children are returned in the order in which they appear in the source code

**Comment by TilakMaddy** [comment]
@alexroan This relies on being able to `GetParentChain` PR  https://github.com/Cyfrin/aderyn/pull/166 .

**Comment by alexroan** [comment]
Closing this as it depends on a PR that has been closed.

**Pull request closed without merging**